vignettes/Data_retrieval.Rmd

```diff
@@ -649,3 +651,146 @@ tcplPlot(lvl = 5, # data level
+
+The following example will consider "Pentachlorophenol" and "liver toxicity"
+
+### Consider *in vivo* toxicity benchmarks as POD-Traditional
+
+First, export ToxRefDB batch download results for any chemical from the CompTox Chemicals Dashboard (https://comptox.epa.gov/dashboard/batch-search) or Hazard APIs (https://api-ccte.epa.gov/docs/) 
+
+After loading all chemical-specific data for "Pentachlorophenol", filter results to only include "liver"-related effects
+```{r txrf-data-liver-subset, eval=FALSE}
+toxref_chnm_liver <- toxref_batch_download_chnm %>% 
+  filter(endpoint_target == 'liver') 
+```
+
+Next identify the observed lowest effect (based on statistically significant i.e. treatment related=1) and lowest observed adverse (i.e. critical) effect levels at minimum dose_adjusted (mg/kg/day) value
+
+```{r txrf-lel-determination, eval=FALSE}
+toxref_chnm_liver_lel<-toxref_chnm_liver %>% 
+  summarise(lel=min(dose_adjusted[treatment_related==1]), 
+            loael=min(dose_adjusted[critical_effect==1]))
+```
+
+### Consider ToxCast *in vitro* bioactivity data as POD-NAM
+
+First, query the invitrodb database for all assay annotations, and filter results to consider only "liver" tissue-based endpoints.
+
+```{r txct-annotations-liver-subset, warning=FALSE, eval=FALSE}
+toxcast_annotations_subset <- tcplQuery("SELECT * FROM assay
+                        INNER JOIN assay_component ON assay.aid = assay_component.aid
+                        INNER JOIN assay_component_endpoint ON assay_component.acid=assay_component_endpoint.acid 
+                        where assay.tissue='liver';")
+```
+
+For this subset of endpoints of targeted interest, pull assay results (mc5-mc6) for the chemical "Pentachlorophenol"
+```{r txct-data-pull, warning=FALSE, message=FALSE, eval=FALSE}
+# Load the chemical data from the database
+chnm <- 'Pentachlorophenol'
+chem <- tcplLoadChem(field = 'chnm',val = chnm)
+                     
+# Load mc5 data from the database for the specified chemical
+mc5 <- tcplLoadData(lvl = 5, # data level 
+                        fld = 'spid', # field to query on
+                        val = chem[,spid], # value for each field (fld)
+                        type = 'mc') # data type - MC
+
+#Join with level 6 flag information
+mc6 <- tcplPrepOtpt(tcplLoadData(lvl=6, fld='m4id', val=mc5$m4id, type='mc'))
+setDT(mc6)
+mc6_mthds <- mc6[ , .( mc6_mthd_id = paste(mc6_mthd_id, collapse=",")), by = m4id]
+mc6_flags <- mc6[ , .( flag = paste(flag, collapse=";")), by = m4id]
+mc5$mc6_flags <- mc6_mthds$mc6_mthd_id[match(mc5$m4id, mc6_mthds$m4id)]
+mc5[, flag.length := ifelse(!is.na(mc6_flags), count.fields(textConnection(mc6_flags), sep =','), NA)]
```

Comment:
This is personal preference, but this line runs longer than the rest. Is it possible to to break this line up into two lines to avoid horizontal scrolling of the code chunk?

Maybe something like this (but with proper indentation):
mc5[, flag.length := ifelse(!is.na(mc6_flags), 
                                         count.fields(textConnection(mc6_flags), sep =','), NA)]